plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -131,7 +141,7 @@ public static HiveMetastore cachingHiveMetastore(HiveMetastore delegate, Executo
 
     public static HiveMetastore cachingHiveMetastore(HiveMetastore delegate, Executor executor, Duration cacheTtl, Optional<Duration> refreshInterval, long maximumSize)
     {
-        if (cacheTtl.toMillis() == 0 || maximumSize == 0) {
+        if (isCacheDisabled(cacheTtl, maximumSize)) {
```

Comment:
check state that cache is enabled

Reply:
Or converted logic to make it checking for enablement.